server/v2/test/user.spec.js

```diff
@@ -0,0 +1,120 @@
+        expect(res.body).to.have.property('message');
+        expect(res.body).to.have.property('data');
+        expect(res.body.status).to.be.eql(201);
+        expect(res.body.data).to.be.an('object');
+        expect(res.body.data).to.have.property('token');
+        done();
+      })
+      .catch((err) => done(err));
+  });
+
+  it('it should not create a user account with invalid email', (done) => {
+    chai.request(server)
+      .post('/api/v2/auth/signup')
+      .send(userData.invalidEmail)
+      .then((res) => {
+        expect(res).to.have.status(401);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(401);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+
+  it('it should not create a user account with an existing email', (done) => {
+    chai.request(server)
+      .post('/api/v2/auth/signup')
+      .send(userData.existingEmail)
+      .then((res) => {
+        expect(res).to.have.status(409);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(409);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+
+  it('it should not create a user account with an existing username', (done) => {
+    chai.request(server)
+      .post('/api/v2/auth/signup')
+      .send(userData.existingUserName)
+      .then((res) => {
+        expect(res).to.have.status(409);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(409);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+
+  it('it should not create a user account with an existing phone number', (done) => {
+    chai.request(server)
+      .post('/api/v2/auth/signup')
+      .send(userData.existingPhoneNumber)
+      .then((res) => {
+        expect(res).to.have.status(409);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(409);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+});
+
+describe('Not Found', async () => {
+  it('it should return 404', (done) => {
+    chai.request(server)
+      .post('/api/v2/auth/sign')
+      .send(userData.noConfirmPassword)
+      .then((res) => {
+        expect(res).to.have.status(404);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('error');
+        expect(res.body.status).to.be.eql(404);
+        done();
+      })
+      .catch((err) => done(err));
+  });
+});
```

Comment:
What is this block testing?

Reply:
Is testing if you didn't entered a valid url

Reply:
And it returns 404